assets/agenda/utils.ts

```diff
@@ -984,75 +984,98 @@ export function isItemTBC(item: IAgendaItem): boolean {
  */
 export function formatCoverageDate(coverage: ICoverage) {
+    if (!coverage.scheduled) {
+        return TO_BE_CONFIRMED_TEXT;
+    }
+
     return get(coverage, TO_BE_CONFIRMED_FIELD) ?
         `${parseDate(coverage.scheduled).format(COVERAGE_DATE_FORMAT)} ${TO_BE_CONFIRMED_TEXT}` :
         parseDate(coverage.scheduled).format(COVERAGE_DATE_TIME_FORMAT);
 }
 
-export const getCoverageTooltip = (coverage: any, beingUpdated?: any, restrictCoverageInfo?: boolean) => {
+const prepareTooltipData = (coverage: any) => {
     const slugline = coverage.item_slugline || coverage.slugline;
     const coverageType = getCoverageDisplayName(coverage.coverage_type);
-    const coverageScheduled = moment(coverage.scheduled);
+    const coverageScheduled = coverage.scheduled ? moment(coverage.scheduled) : null;
     const assignee = getCoverageAsigneeName(coverage);
     const desk = getCoverageDeskName(coverage);
     const assignedDetails = [
         assignee ? gettext('assignee: {{name}}', {name: assignee}) : '',
         desk ? gettext('desk: {{name}}', {name: desk}) : '',
     ].filter((x) => x !== '').join(', ');
 
-    if (coverage.coverage_status !== COVERAGE_INTENDED) {
-        return get(DRAFT_STATUS_TEXTS, coverage.coverage_status, '');
-    } else if (coverage.workflow_status === WORKFLOW_STATUS.DRAFT) {
+    return {slugline, coverageType, coverageScheduled, assignedDetails};
+};
+
+const handleStatus = (coverage: any, data: any, beingUpdated?: any, restrictCoverageInfo?: boolean) => {
+    const {slugline, coverageType, coverageScheduled, assignedDetails} = data;
+    const {workflow_status} = coverage;
+
+    switch (workflow_status) {
+    case WORKFLOW_STATUS.DRAFT:
         return gettext('{{ type }} coverage {{ slugline }} {{ status_text }} {{assignedDetails}}', {
             type: coverageType,
             slugline: slugline,
             status_text: getCoverageStatusText(coverage),
             assignedDetails,
         });
-    } else if (coverage.workflow_status === WORKFLOW_STATUS.ASSIGNED) {
+
+    case WORKFLOW_STATUS.ASSIGNED:
         if (restrictCoverageInfo) {
             return gettext('Planned {{ type }} coverage {{ slugline }} {{assignedDetails}}', {
                 type: coverageType,
                 slugline: slugline,
                 assignedDetails,
             });
-        } else {
-            return gettext('Planned {{ type }} coverage {{ slugline }}, expected {{date}} at {{time}} {{assignedDetails}}', {
+        }
+        return coverageScheduled ? 
+            gettext('Planned {{ type }} coverage {{ slugline }}, expected {{date}} at {{time}} {{assignedDetails}}', {
                 type: coverageType,
                 slugline: slugline,
                 date: formatDate(coverageScheduled),
                 time: formatTime(coverageScheduled),
                 assignedDetails,
+            }) :
+            gettext('Planned {{ type }} coverage {{ slugline }} {{assignedDetails}}', {
```

Comment:
There seems to be lots of duplication of text in these cases, between `restrictCoverageInfo` and `coverageScheduled`.

Maybe we can use a single variable before the `switch` statement or'ing between these two, and use that in each case block